ue4docker/build.py

```diff
@@ -363,91 +365,99 @@ def build():
+
+                prereqConsumerArgs = [
+                    "--build-arg",
+                    "PREREQS_TAG={}".format(config.prereqsTag),
+                ]
+            else:
+                logger.info("Skipping ue4-build-prerequisities image build.")
 
             # Build the UE4 source image
-            prereqConsumerArgs = [
-                "--build-arg",
-                "PREREQS_TAG={}".format(config.prereqsTag),
-            ]
-            credentialArgs = [] if len(secrets) > 0 else endpoint.args()
-            ue4SourceArgs = prereqConsumerArgs + [
-                "--build-arg",
-                "GIT_REPO={}".format(config.repository),
-                "--build-arg",
-                "GIT_BRANCH={}".format(config.branch),
-                "--build-arg",
-                "VERBOSE_OUTPUT={}".format("1" if config.verbose == True else "0"),
-            ]
-            builder.build(
-                "ue4-source",
-                mainTags,
-                commonArgs + config.platformArgs + ue4SourceArgs + credentialArgs,
-                secrets,
-            )
-            builtImages.append("ue4-source")
+            if config.buildTargetSource:
+                # Start the HTTP credential endpoint as a child process and wait for it to start
+                if config.opts.get("credential_mode", "endpoint") == "endpoint":
+                    endpoint = CredentialEndpoint(username, password)
+                    endpoint.start()
+
+                # If we're using build secrets then pass the Git username and password to the UE4 source image as secrets
+                secrets = {}
+                if config.opts.get("credential_mode", "endpoint") == "secrets":
+                    secrets = {"username": username, "password": password}
+                credentialArgs = [] if len(secrets) > 0 else endpoint.args()
+
+                ue4SourceArgs = prereqConsumerArgs + [
+                    "--build-arg",
+                    "GIT_REPO={}".format(config.repository),
+                    "--build-arg",
+                    "GIT_BRANCH={}".format(config.branch),
+                    "--build-arg",
+                    "VERBOSE_OUTPUT={}".format("1" if config.verbose == True else "0"),
+                ]
+                builder.build(
+                    "ue4-source",
+                    mainTags,
+                    commonArgs + config.platformArgs + ue4SourceArgs + credentialArgs,
+                    secrets,
+                )
+                builtImages.append("ue4-source")
+            else:
+                logger.info("Skipping ue4-source image build.")
 
             # Build the UE4 Engine source build image, unless requested otherwise by the user
-            ue4BuildArgs = prereqConsumerArgs + [
-                "--build-arg",
-                "TAG={}".format(mainTags[1]),
-            ]
-            if config.noEngine == False:
+            if config.buildTargetEngine:
```

Comment:
This condition is wrong. `ue4BuildArgs` are used below on line 453 to build `ue4-minimal`.

Reply:
Good catch! Fixed, I think - let me know if it looks right.